### PR TITLE
Move ShpTDLoader, LZO and XORDelta formats to Mods.Cnc

### PR DESCRIPTION
**Pull request by reaperrr**
They're pretty much RA/TD-specific formats.

Another (and presumably the last) split from #17355.

### CODE REVIEW

File: OpenRA.Mods.Cnc/FileFormats/LZOCompression.cs [hunk 1]
```diff
@@ -67,7 +67,7 @@
 
 using System;
 
-namespace OpenRA.Mods.Common.FileFormats
```
Comment (Mailaender) [inline OpenRA.Mods.Cnc/FileFormats/LZOCompression.cs]: https://de.wikipedia.org/wiki/Lempel-Ziv-Oberhumer is probably a generic algorithm.

Comment (reaperrr) [reply OpenRA.Mods.Cnc/FileFormats/LZOCompression.cs]: I guess so, but it's only used by ShpTD currently.
@pchote what do you think?

Comment (pchote) [reply OpenRA.Mods.Cnc/FileFormats/LZOCompression.cs]: If its not used by anything in `Mods.Common`, and theres no obvious need for it to be used by anything in `Mods.Common` in the future then it doesn't belong in `Mods.Common`.

Comment (GraionDilach) [reply OpenRA.Mods.Cnc/FileFormats/LZOCompression.cs]: Uhm...

https://github.com/OpenRA/OpenRA/blob/4dd08d9dc2d4f8bf1a6407807211780e156befac/OpenRA.Mods.Cnc/UtilityCommands/ImportTSMapCommand.cs#L295 is the sole usecase currently.

TD SHPs use LCW, not LZO.

Comment (GraionDilach) [reply OpenRA.Mods.Cnc/FileFormats/LZOCompression.cs]: FWIW, LZO could've been added as a common package next to zip based on this.

Comment (reaperrr) [reply OpenRA.Mods.Cnc/FileFormats/LZOCompression.cs]: > TD SHPs use LCW, not LZO.

Yeah, guess i was too tired that day.

> FWIW, LZO could've been added as a common package next to zip based on this.

I'm not in the mood to get my head around that, let alone spend the time on it. And we can still move it back whenever anyone - whether us or some 3rd-party contributor - finds the time and motivation to tackle this.

**Review by teinarss — DISMISSED** [review]

File: OpenRA.Mods.Cnc/SpriteLoaders/ShpD2Loader.cs [hunk 2]
```diff
@@ -12,6 +12,7 @@
 using System;
 using System.IO;
 using OpenRA.Graphics;
+using OpenRA.Mods.Cnc.FileFormats;
```
Comment (pchote) [inline OpenRA.Mods.Cnc/SpriteLoaders/ShpD2Loader.cs]: This doesn't seem to be used?

File: OpenRA.Mods.Cnc/UtilityCommands/ImportRedAlertLegacyMapCommand.cs [hunk 3]
```diff
@@ -14,6 +14,7 @@
 using System.IO;
 using System.Linq;
 using System.Text;
+using OpenRA.Mods.Cnc.FileFormats;
```
Comment (pchote) [inline OpenRA.Mods.Cnc/UtilityCommands/ImportRedAlertLegacyMapCommand.cs]: This doesn't seem to be used?

File: OpenRA.Mods.Cnc/UtilityCommands/ImportTiberianDawnLegacyMapCommand.cs [hunk 4]
```diff
@@ -13,6 +13,7 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using OpenRA.Mods.Cnc.FileFormats;
```
Comment (pchote) [inline OpenRA.Mods.Cnc/UtilityCommands/ImportTiberianDawnLegacyMapCommand.cs]: This doesn't seem to be used?

**Comment by reaperrr** [comment]
Fixed.

**Review by pchote — APPROVED** [review]

**Pull request merged**